packages/cli-e2e/__tests__/atomic.specs.ts

```diff
@@ -77,182 +88,205 @@ describe('ui:create:atomic', () => {
-      'src/components/sample-component/sample-component.css',
-      'src/components/sample-result-component/sample-result-component.tsx',
-      'src/components/sample-result-component/sample-result-component.css',
-    ];
-
-    test.each(createdFilesPaths)(
-      'should create the %s file or directory',
-      (path) => expect(projectFileExist(path)).toBe(true)
-    );
-
-    const deletedFilesPaths = [
-      'scripts/clean-up.js',
-      'scripts/setup-lamdba.js',
-      'scripts/utils.js',
-    ];
-
-    test.each(deletedFilesPaths)(
-      'should delete the %s file or directory',
-      (path) => expect(projectFileExist(path)).toBe(false)
-    );
-  });
-
-  afterEach(async () => {
-    await captureScreenshots(browser);
-  });
-
-  afterAll(async () => {
-    process.env = oldEnv;
-    await browser.close();
-    await Promise.all(
-      processManagers.map((manager) => manager.killAllProcesses())
-    );
-  });
-
-  describe('when the project is configured correctly', () => {
-    let serverProcessManager: ProcessManager;
-    let interceptedRequests: HTTPRequest[] = [];
-    let consoleInterceptor: BrowserConsoleInterceptor;
-
+  function setupTests(options: BuildAppOptions) {
     beforeAll(async () => {
-      serverProcessManager = new ProcessManager();
-      processManagers.push(serverProcessManager);
-      const appTerminal = await startApplication(
-        serverProcessManager,
-        'atomic-server-valid'
+      await loginWithApiKey(
+        process.env.PLATFORM_API_KEY!,
+        process.env.ORG_ID!,
+        process.env.PLATFORM_ENV!
       );
-      await waitForAppRunning(appTerminal);
-    }, 5 * 60e3);
+      const processManager = new ProcessManager();
+      processManagers.push(processManager);
+      browser = await getNewBrowser();
+      await buildApplication(processManager, options);
+      await processManager.killAllProcesses();
+    }, 15 * 60e3);
 
     beforeEach(async () => {
-      consoleInterceptor = new BrowserConsoleInterceptor(page, projectName);
-      await consoleInterceptor.startSession();
-
-      page.on('request', (request: HTTPRequest) => {
-        interceptedRequests.push(request);
-      });
+      jest.resetModules();
+      process.env = {...oldEnv};
+      page = await openNewPage(browser, page);
     });
 
     afterEach(async () => {
-      page.removeAllListeners('request');
-      interceptedRequests = [];
-      await consoleInterceptor.endSession();
+      await captureScreenshots(browser);
     });
 
     afterAll(async () => {
-      await serverProcessManager.killAllProcesses();
-    }, 5 * 30e3);
+      process.env = oldEnv;
+      await browser.close();
+      await Promise.all(
+        processManagers.map((manager) => manager.killAllProcesses())
+      );
+    });
+  }
 
-    it('should not contain console errors nor warnings', async () => {
-      await page.goto(searchPageEndpoint, {
-        waitUntil: 'networkidle2',
-      });
+  function runTests() {
```

Comment:
Wrapping `test`, `describe` and `expect` in a function is a no-go, it can cause issues with Jest custom parsing sauce and lead, at best, to decrease performance, at worst, false positive/negative.
I'll get another look very soon to help you refactor that proper ;)

Reply:
It's mostly OK when everything is 'in band' but for example, Jest is adding sharding in v28, which would be a boon for us to have ootb support to run tests on multiple GHA runners (we're doing it currently, but hackish way).